### PR TITLE
Compact the calendar view with controls embedded in the top menu bar

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Compact the calendar view so that there is more vertical screen real estate. This puts the controls embedded in the top menu bar similar to the energy panel.

The approach is to pull the navigation (Next, Previous, Today and the date range label) as well as the date view controls into a separate component. This is pulled out of full calendar into the caller panel so that it can be tied into the action item slot of the menu bar. The panel and card now invoke it and set different display options depending on the screen size. (part of the controls spill out of the menu bar when in narrow mode). The display changes a bit to look more like the energy panel as a result, though that is not necessarily a goal.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

Full screen view:
Screenshot 2023-09-08 at 8 52 14 AM

Narrow / Mobile view:
Screenshot 2023-09-08 at 8 52 29 AM

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/calendar/ha-full-calendar.ts [hunk 1]
```diff
@@ -342,74 +283,15 @@ export class HAFullCalendar extends LitElement {
...
-  private _handleToday(): void {
-    this.calendar!.today();
-    this._fireViewChanged();
-  }
-
-  private _handleView(ev: CustomEvent): void {
-    this._activeView = ev.detail.value;
-    this.calendar!.changeView(this._activeView!);
-    this._fireViewChanged();
-  }
```
Comment (bramkragten) [inline src/panels/calendar/ha-full-calendar.ts:364]: Make this public functions

File: src/panels/calendar/ha-calendar-app-bar.ts [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  private _handleRefresh() {
+    fireEvent(this, "refresh");
+  }
+
+  private _handleSelectView(ev: CustomEvent): void {
+    ev.stopPropagation();
+    if (this.params.selectView) {
+      this.params.selectView(ev.detail.value);
+    }
+  }
```
Comment (bramkragten) [inline src/panels/calendar/ha-calendar-app-bar.ts:145]: make these fire events

Comment (bramkragten) [reply src/panels/calendar/ha-calendar-app-bar.ts:145]: You can create a single helper function that maps the events to the function calls

Comment (allenporter) [reply src/panels/calendar/ha-calendar-app-bar.ts]: Thanks, fixed. I did have it this way (looks like i left the events already in there!) but then didn't like the manual wiring up of different callbacks. The single helper seems better. I ended up copying between the calendar card and panel which may be ok, but let me know if its worth putting that function somewhere common (i'm not sure where to put it)

**Comment by allenporter** [comment]
Is this UI change worth proceeding with if I were to resolve the conflicts? I think it needs an entire rewrite since this code was refactored to support todos, but want to make sure we want this before I take it on.

**Comment by bramkragten** [comment]
There is a design for it, I was planning to pick this up myself, but if you want to give it crack, you are more than welcome!

**Comment by allenporter** [comment]
Did the design look similar to the screenshot above? Want to make sure I know what the desired end state is.

I'd say this is in my backlog at  low priority, but just didn't want to leave it hanging... happy for whoever gets to it first

**Pull request closed without merging**